ShellPkg/Library/UefiShellDebug1CommandsLib/Edit/Edit.c

```diff
@@ -101,8 +101,14 @@ ShellCommandRunEdit (
         //
         if (ShellCommandLineGetCount (Package) == 2) {
           TempParam = ShellCommandLineGetRawValue (Package, 1);
-          ASSERT (TempParam != NULL);
-          FileBufferSetFileName (TempParam);
+          if (TempParam == NULL) {
+            ASSERT (TempParam != NULL);
+            ShellPrintHiiEx (-1, -1, NULL, STRING_TOKEN (STR_GEN_TOO_MANY), gShellDebug1HiiHandle, L"edit");

```

Comment:
Maybe another error print than `STR_GEN_TOO_MANY`

Reply:
Changed to STR_EDIT_MAIN_INIT_FAILED